ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/PrivacyController.java

```diff
@@ -74,17 +74,21 @@ long determineEeaNonce(
   Optional<String> buildAndSendAddPayload(
       PrivateTransaction privateTransaction, Bytes32 privacyGroupId, String enclaveKey);
 
-  Optional<PrivacyGroup> retrieveOffChainPrivacyGroup(String toBase64String, String enclaveKey);
+  Optional<PrivacyGroup> findOffChainPrivacyGroupByGroupId(
+      String toBase64String, String enclaveKey);
 
-  List<PrivacyGroup> findOnChainPrivacyGroup(List<String> asList, String enclaveKey);
+  Optional<PrivacyGroup> findPrivacyGroupByGroupId(
```

Comment:
should this be findOnChainPrivacyGroupByGroupId?

Reply:
See reply from earlier. The `findOnChainPrivacyGroupByGroupId` is in the implementation, not the interface.

Reply:
👍